pkg/executor/join/hash_join_v2.go

```diff
@@ -419,29 +826,50 @@ func (e *HashJoinV2Exec) initializeForProbe() {
 	}
 }
 
-func (e *HashJoinV2Exec) fetchAndProbeHashTable(ctx context.Context) {
-	e.initializeForProbe()
-	fetchProbeSideChunksFunc := func() {
-		defer trace.StartRegion(ctx, "HashJoinProbeSideFetcher").End()
-		e.ProbeSideTupleFetcher.fetchProbeSideChunks(
-			ctx,
-			e.MaxChunkSize(),
-			func() bool { return e.ProbeSideTupleFetcher.hashTableContext.hashTable.isHashTableEmpty() },
-			e.ProbeSideTupleFetcher.canSkipProbeIfHashTableIsEmpty,
-			e.ProbeSideTupleFetcher.needScanRowTableAfterProbeDone,
-			e.ProbeSideTupleFetcher.shouldLimitProbeFetchSize(),
-			&e.ProbeSideTupleFetcher.hashJoinCtxBase)
-	}
-	e.workerWg.RunWithRecover(fetchProbeSideChunksFunc, e.ProbeSideTupleFetcher.handleProbeSideFetcherPanic)
+func (e *HashJoinV2Exec) startProbeFetcher(ctx context.Context) {
+	if !e.inRestore {
+		fetchProbeSideChunksFunc := func() {
+			defer trace.StartRegion(ctx, "HashJoinProbeSideFetcher").End()
+			e.ProbeSideTupleFetcher.fetchProbeSideChunks(
+				ctx,
+				e.MaxChunkSize(),
+				func() bool { return e.ProbeSideTupleFetcher.hashTableContext.hashTable.isHashTableEmpty() },
+				func() bool { return e.spillHelper.isSpillTriggered() },
+				e.ProbeSideTupleFetcher.canSkipProbeIfHashTableIsEmpty,
+				e.ProbeSideTupleFetcher.needScanRowTableAfterProbeDone,
+				e.ProbeSideTupleFetcher.shouldLimitProbeFetchSize(),
+				&e.ProbeSideTupleFetcher.hashJoinCtxBase)
+		}
+		e.workerWg.RunWithRecover(fetchProbeSideChunksFunc, e.ProbeSideTupleFetcher.handleProbeSideFetcherPanic)
+	}
+}
+
```

Comment:
Would this be more clear?

```suggestion
func (e *HashJoinV2Exec) fetchAndProbeHashTable(ctx context.Context) {
	if e.inRestore {
		if err := <-e.buildFinished; err != nil {
			return
		}
		e.startRestoreProbeJoinWorkers(ctx)
	} else {
		e.startProbeFetcher(ctx)
		e.startNormalProbeJoinWorkers(ctx)
	}
	e.waiterWg.RunWithRecover(e.waitJoinWorkers, nil)
}

func (e *HashJoinV2Exec) startProbeFetcher(ctx context.Context) {
	fetchProbeSideChunksFunc := func() {
		defer trace.StartRegion(ctx, "HashJoinProbeSideFetcher").End()
		e.ProbeSideTupleFetcher.fetchProbeSideChunks(
			ctx,
			e.MaxChunkSize(),
			e.ProbeSideTupleFetcher.hashTableContext.hashTable.isHashTableEmpty,
			e.spillHelper.isSpillTriggered,
			e.ProbeSideTupleFetcher.canSkipProbeIfHashTableIsEmpty,
			e.ProbeSideTupleFetcher.needScanRowTableAfterProbeDone,
			e.ProbeSideTupleFetcher.shouldLimitProbeFetchSize(),
			&e.ProbeSideTupleFetcher.hashJoinCtxBase,
		)
	}
	e.workerWg.RunWithRecover(fetchProbeSideChunksFunc, e.ProbeSideTupleFetcher.handleProbeSideFetcherPanic)
}

func (e *HashJoinV2Exec) startNormalProbeJoinWorkers(ctx context.Context) {
	for i := uint(0); i < e.Concurrency; i++ {
		workerID := i
		e.workerWg.RunWithRecover(func() {
			defer trace.StartRegion(ctx, "HashJoinWorker").End()
			e.ProbeWorkers[workerID].runJoinWorker()
		}, e.ProbeWorkers[workerID].handleProbeWorkerPanic)
	}
}

func (e *HashJoinV2Exec) startRestoreProbeJoinWorkers(ctx context.Context) {
	for i := uint(0); i < e.Concurrency; i++ {
		workerID := i
		e.workerWg.RunWithRecover(func() {
			defer trace.StartRegion(ctx, "HashJoinWorker").End()
			e.ProbeWorkers[workerID].restoreAndProbe(e.restoredProbeInDisk[workerID])
		}, e.ProbeWorkers[workerID].handleProbeWorkerPanic)
	}
}

```

Reply:
I think these two ways of writing are same.